src/MatrixClientPeg.js

```diff
@@ -40,6 +48,14 @@ function deviceId() {
 class MatrixClientPeg {
     constructor() {
         this.matrixClient = null;
+
+        // These are the default options used when Lifecycle.js
+        // starts the client. These can be altered at any
+        // time up to after the 'will_start_client' event is
+        // finished processing.
+        this.opts = {
```

Comment:
So I see your point about keeping global state out of Lifecycle.js, but having Lifecycle.js yank it out of here just doesn't feel right.

Suggestion: wrap `MatrixClient.start` with a function in `MatrixClientPeg`, and populate the options there.